app/jobs/subscription_placement_job.rb

```diff
@@ -4,9 +4,10 @@
 
 class SubscriptionPlacementJob < ActiveJob::Base
   def perform
-    ids = proxy_orders.pluck(:id)
-    ProxyOrder.where(id: ids).each do |proxy_order|
-      place_order_for(proxy_order)
+    ProxyOrder.transaction do
+      while proxy_order = proxy_orders.lock.first do
```

Comment:
What was the motivation behind using `while` here? Wouldn't `proxy_orders.lock.each` be more idiomatic Ruby?

Reply:
Hmmm... my local Rspec process seems to consistently hang when it reaches this while loop.

Reply:
I read that `lock.each` doesn't guarantee in which order the records are locked which can lead to a deadlock. So I thought if I have a deterministic order and lock them one by one then it would be safer in avoiding deadlocks. But I'm reconsidering that now. `while` is almost never a good idea. I have seen tutorials where the `order by` clause in Postgresql is supposed to determine the order of locking but I'm not sure that it's true. Considering that we run this job only every five minutes, we probably don't have to worry about deadlocks at this stage. If it's a problem, the spec will become flaky.